src/api/index.ts

```diff
@@ -120,13 +118,15 @@ function init(dbPath: string): Context {
 
 const ApiPlugin: FastifyPluginAsync<MapServerOptions> = async (
   fastify,
-  { dbPath }
+  { database }
 ) => {
-  if (dbPath == null)
-    throw new Error('Map server option `dbPath` must be specified')
+  if (database == null || !(database instanceof Db))
```

Comment:
I think this might be unreliable as a check. Because it is now a peer dependency, it is possible that an app using this code could create the instance with a different version of better-sqlite3 than the one that resolves here. Also requiring it in this file makes it not really a peer dependency (because this will break if it is not installed). I think we need to lean on typescript for checking, and it we really want to check then maybe check a property or method? However if it is not a BetterSqlite3.Database instance then something will throw later on pretty quick too.

Reply:
ah yes true! totally overlooked that. will remove the `instanceOf` check but not sure i can think of a better check, so maybe okay to omit and let things break in that case

Reply:
> Also requiring it in this file makes it not really a peer dependency (because this will break if it is not installed)

does this mean that I can't import the TS type from it either? guessing the answer is yes

Reply:
ah looks like this may be what i'm looking for: https://stackoverflow.com/questions/54392809/how-do-i-handle-optional-peer-dependencies-when-publishing-a-typescript-package/70096366#70096366

seems to work as expected when trying locally via npm pack